pkg/ring/kv/dynamodb/dynamodb.go

```diff
@@ -161,6 +162,55 @@ func (kv dynamodbKV) Delete(ctx context.Context, key dynamodbKey) error {
 }
 
 func (kv dynamodbKV) Put(ctx context.Context, key dynamodbKey, data []byte) error {
+	input := &dynamodb.PutItemInput{
+		TableName: kv.tableName,
+		Item:      kv.generatePutItemRequest(key, data),
+	}
+	_, err := kv.ddbClient.PutItemWithContext(ctx, input)
+	return err
+}
+
+func (kv dynamodbKV) Batch(ctx context.Context, put map[dynamodbKey][]byte, delete []dynamodbKey) error {
+	var writeRequests []*dynamodb.WriteRequest
```

Comment:
One small nit: we can allocate slice with size

Reply:
makes sense. we know the size